### PR TITLE
fix(upgrade): change type of local-rke-state secret

### PR DESCRIPTION
**Pull request by starbops**
**Problem:**

The `local-rke-state` Secret's type is originally `Opaque`. After Rancher v2.7, it needs to be explicitly set to the type of `rke.cattle.io/cluster-state`. Otherwise, the following node upgrade (RKE2 upgrade) will be blocked by the check.

**Solution:**

Do the type conversion at the "apply manifest" phase. After Rancher is upgraded, we scale down the Rancher deployment to 0, change the type of the `local-rke-state` Secret, update it to the cluster, and finally bring back the Rancher deployment and continue the remaining steps.

**Related Issue:**

#3858 

**Test plan:**

1. Prepare a v1.1.2 Harvester cluster (if don't need to validate along with the multi-cordon issue, a single node cluster would be fine)
2. Upgrade with the master ISO image with this fix included
3. Check the type of the `local-rke-state` Secret after apply-manifest Job is done. It should be the type of `rke.cattle-.io/cluster-state`
```
$ kubectl -n fleet-local get secrets local-rke-state -o yaml
apiVersion: v1
data:
  agentToken: dGVzdA==
  serverToken: dGVzdA==
kind: Secret
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"agentToken":"dGVzdA==","serverToken":"dGVzdA=="},"kind":"Secret","metadata":{"annotations":{},"creationTimestamp":"2023-06-08T02:15:20Z","name":"local-rke-state","namespace":"fleet-local","resourceVersion":"2049","uid":"b80c25ec-b211-4c48-b743-e7122887be44"},"type":"rke.cattle.io/current-state"}
  creationTimestamp: "2023-06-08T07:41:20Z"
  name: local-rke-state
  namespace: fleet-local
  resourceVersion: "321554"
  uid: 1c7c4e2d-ebd8-43ce-95c5-a82b2253f6d9
type: rke.cattle.io/current-state
```
4. The upgrade should be successful

### CODE REVIEW

File: package/upgrade/upgrade_manifests.sh [hunk 1]
```diff
@@ -278,6 +278,36 @@ upgrade_rancher() {
...
+  # Starting from Rancher v2.7, the local-rke-state Secret needs to be in type of "rke.cattle.io/current-state"
+  # Need to convert it from the "Opaque" type; otherwise, the following RKE2 upgrades won't start
+  # Ref: https://github.com/rancher/rancher/pull/41088
+  echo "Check current local-rke-state secret type..."
+  current_secret=$(mktemp -p $UPGRADE_TMP_DIR)
+  kubectl -n fleet-local get secrets local-rke-state -o yaml > $current_secret
+  current_secret_type=$(yq '.type' $current_secret)
+  if [ "$current_secret_type" = "Opaque" ]; then
+    new_secret=$(mktemp -p $UPGRADE_TMP_DIR)
+    sed 's,type: Opaque,type: rke.cattle.io/cluster-state,g' $current_secret > $new_secret
```
Comment (bk201) [inline package/upgrade/upgrade_manifests.sh]: can we use `yq` to modify the value :)

Comment (starbops) [reply package/upgrade/upgrade_manifests.sh]: updated, thanks!

**Review by bk201 — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Comment by guangbochen** [comment]
@mergify backport v1.2

**Pull request merged**